source/agora/node/Ledger.d

```diff
@@ -326,24 +272,31 @@ public class Ledger
 
         Params:
             block = the block to add
+            externalized = whether the block is `externalized` now
 
         Returns:
             true if the block was accepted
 
     ***************************************************************************/
 
-    public bool acceptBlock (const ref Block block,
+    public bool acceptBlock (const ref Block block, bool externalized = true,
         string file = __FILE__, size_t line = __LINE__) @safe
     {
         if (auto fail_reason = block.header.height == 0
             ? block.isGenesisBlockInvalidReason()
             : this.validateBlock(block, file, line))
         {
-            log.trace("Rejected block: {}: {}", fail_reason, block.prettify());
-            return false;
+            if (externalized)
+            {
+                log.trace("Rejected block: {}: {}", fail_reason, block.prettify());
+                return false;
+            }
+            else
+                throw new Exception("A block loaded from disk is invalid: " ~
+                    fail_reason);
```

Comment:
Why not `return false` here and throw in the constructor ?
Having a function that `throw` or return `false` depending on a boolean parameter is usually not a good idea.

Reply:
The loading process of the constructor was throwing `Exception`. But, `return false` would be more consistent because the original `acceptBlock` just returned a boolean value.